tests/pkg/asserters/notifications.go

```diff
@@ -0,0 +1,94 @@
+	appNamespace                       string
+	region                             string
+	tenant                             string
+	tenantParentCustomer               string
+	externalServicesMockMtlsSecuredURL string
+	client                             *http.Client
+}
+
+func NewNotificationsAsserter(expectedNotificationsCount int, op string, targetObjectID, sourceObjectID string, localTenantID string, appNamespace string, region string, tenant string, tenantParentCustomer string, externalServicesMockMtlsSecuredURL string, client *http.Client) *NotificationsAsserter {
+	return &NotificationsAsserter{expectedNotificationsCount: expectedNotificationsCount, op: op, targetObjectID: targetObjectID, sourceObjectID: sourceObjectID, localTenantID: localTenantID, appNamespace: appNamespace, region: region, tenant: tenant, tenantParentCustomer: tenantParentCustomer, externalServicesMockMtlsSecuredURL: externalServicesMockMtlsSecuredURL, client: client}
+}
+
+func (a *NotificationsAsserter) AssertExpectations(t *testing.T, ctx context.Context) {
+	formationID := ctx.Value(context_keys.FormationIDKey).(string)
+
+	body := getNotificationsFromExternalSvcMock(t, a.client, a.externalServicesMockMtlsSecuredURL)
+	assertNotificationsCount(t, body, a.targetObjectID, a.expectedNotificationsCount)
+
+	notificationsForTarget := gjson.GetBytes(body, a.targetObjectID)
+	assignNotificationAboutSource := notificationsForTarget.Array()[0]
+	assertFormationAssignmentsNotificationWithConfigContainingItemsStructure(t, assignNotificationAboutSource, assignOperation, formationID, a.sourceObjectID, a.localTenantID, a.appNamespace, a.region, a.tenant, a.tenantParentCustomer, nil)
+}
+
+func getNotificationsFromExternalSvcMock(t *testing.T, client *http.Client, ExternalServicesMockMtlsSecuredURL string) []byte {
+	t.Logf("Getting formation notifications recieved in external services mock")
+	resp, err := client.Get(ExternalServicesMockMtlsSecuredURL + "/formation-callback")
+	defer func() {
+		if err := resp.Body.Close(); err != nil {
+			t.Logf("Could not close response body %s", err)
+		}
+	}()
+	require.NoError(t, err)
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Equal(t, http.StatusOK, resp.StatusCode, fmt.Sprintf("actual status code %d is different from the expected one: %d. Reason: %v", resp.StatusCode, http.StatusOK, string(body)))
+	return body
+}
+
+func assertNotificationsCount(t *testing.T, body []byte, objectID string, count int) {
+	notifications := gjson.GetBytes(body, objectID)
+	if count > 0 {
+		require.True(t, notifications.Exists())
+		require.Len(t, notifications.Array(), count)
+	} else {
+		require.False(t, notifications.Exists())
+	}
+}
+
+func assertFormationAssignmentsNotificationWithConfigContainingItemsStructure(t *testing.T, notification gjson.Result, op, formationID, expectedAppID, expectedLocalTenantID, expectedAppNamespace, expectedAppRegion, expectedTenant, expectedCustomerID string, expectedConfig *string) {
```

Comment:
I see we're using the old "items structure" which at this point I think maybe it's okay because all of the tests are using it(except a few ones), however, do you think it will be helpful to have the other assert that is using the new structure as well? We'll have another improvement task to simplify and make it easier using different webhook input templates within the current abstraction that you made but if we put here the "old items structure" my concern is that the new tests that will be written will be using the old structure and after that, we need to refactor them as well. At least the new tests have the new structure and regarding the old ones, we'll gradually migrate them to the new abstraction and the new webhook structure?

Reply:
Will be adapted in a followup PR